http/handler/handler.go

```diff
@@ -91,12 +91,12 @@ func (h *Handler) ServeHTTP(rw http.ResponseWriter, request *http.Request) {
 		return
 	}
 
-	if !h.securedProtocol && !h.edgeManager.IsKeySet() {
+	if !h.securedProtocol && !(h.edgeManager != nil && h.edgeManager.IsKeySet()) {
```

Comment:
Agree with the point above, it might be better to pass this check as some form of parameter for readability.